src/main/java/com/redhat/red/build/koji/kerberos/KrbUtils.java

```diff
@@ -0,0 +1,254 @@
+import org.apache.kerby.kerberos.kerb.client.KrbConfig;
+import org.apache.kerby.kerberos.kerb.client.KrbOption;
+import org.apache.kerby.kerberos.kerb.common.EncryptionUtil;
+import org.apache.kerby.kerberos.kerb.type.EncKrbPrivPart;
+import org.apache.kerby.kerberos.kerb.type.ap.ApRep;
+import org.apache.kerby.kerberos.kerb.type.ap.ApReq;
+import org.apache.kerby.kerberos.kerb.type.ap.Authenticator;
+import org.apache.kerby.kerberos.kerb.type.ap.EncAPRepPart;
+import org.apache.kerby.kerberos.kerb.type.base.EncryptionKey;
+import org.apache.kerby.kerberos.kerb.type.base.KeyUsage;
+import org.apache.kerby.kerberos.kerb.type.base.KrbMessageType;
+import org.apache.kerby.kerberos.kerb.type.base.NameType;
+import org.apache.kerby.kerberos.kerb.type.base.PrincipalName;
+import org.apache.kerby.kerberos.kerb.type.ticket.TgtTicket;
+
+public class KrbUtils
+{
+    public static String makeServerPrincipal( String service, String url, String realm )
+            throws UnknownHostException, MalformedURLException
+    {
+        String serverName = InetAddress.getByName( new URL( url ).getHost() ).getCanonicalHostName();
+
+        String serverPrincipal = new PrincipalName( service + "/" + serverName + "@" + realm, NameType.NT_UNKNOWN ).toString();
+
+        return serverPrincipal;
+    }
+
+    public static String getKrb5ConfFilename()
+    {
+        String filename = System.getProperty( "java.security.krb5.conf" );
+
+        if ( filename == null )
+        {
+            filename = System.getProperty( "java.home" ) + File.separator + "lib" + File.separator + "security" + File.separator + "krb5.conf";
+
+            Path javaHomeLibSecurityPath = Paths.get( filename );
+
+            if ( !Files.exists( javaHomeLibSecurityPath, new LinkOption[] { } ) )
+            {
+                String osName = System.getProperty( "os.name" );
+
+                if ( osName.startsWith( "Windows" ) )
+                {
+                    String systemRoot = System.getenv( "SYSTEMROOT" );
+
+                    if ( systemRoot != null )
+                    {
+                        filename = systemRoot + File.separator + "krb5.ini";
+
+                    }
+                }
+                else if ( osName.startsWith( "SunOS" ) )
+                {
+                    filename = File.separator + "etc" + File.separator + "krb5" + File.separator + "krb5.conf";
+                }
+                else
+                {
+                    filename = File.separator + "etc" + File.separator + "krb5.conf";
+                }
+            }
+        }
+
+        Path filenamePath = Paths.get( filename );
+
+        if ( !Files.exists( filenamePath, new LinkOption[] { } ) || !Files.isReadable( filenamePath ) )
+        {
+            filename = null;
+        }
+
+        return filename;
+    }
+
+    public static KrbClient newClient( String krb5ConfFilename )
+            throws IOException, KrbException {
+        KrbConfig krbConfig = new KrbConfig();
```

Comment:
As you an probably tell, I had originally meant for `KrbUtils` to just have a couple static methods like `makeServerPrincipal` and `getKrb5ConfFilename` that shouldn't go in `KojiClient`.  Later, I started pulling most of the actual kerberos code out of the `KojiClient` class and putting it in `KrbUtils` as well, which lead to these methods with seemingly long list of parameters.

But, what did you have in mind by turning it into a regular component to encapsulate the Kerberos stuff? Or are we just moving these methods into `KrbAuthenticator`? I can look into how to improve these method signatures, in any case.